IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m

```diff
@@ -123,8 +123,14 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
         if (enrollmentId) return enrollmentId;
     }
     
-    // If we haven't found an exact match yet, fallback to any enrollment ID to support no userID or single userID scenarios.
-    return [self enrollmentIdIfAvailableWithContext:context error:error];
+    // If we haven't found an exact match yet, fallback to any enrollment ID to support no userID or single userID scenarios
+    // only if homeAccountId or legacyId are not empty
+    if (![NSString msidIsStringNilOrBlank:homeAccountId] || ![NSString msidIsStringNilOrBlank:legacyUserId])
```

Comment:
Maybe not to return 1st enrollmentId unless both homeAccountId and userId are empty?

```if ([NSString msidIsStringNilOrBlank:homeAccountId] && [NSString msidIsStringNilOrBlank:legacyUserId])```

Reply:
Thank you Kai, I got the logic backwards 😵‍💫

Reply:
Thanks for updating!